### PR TITLE
Add support for Java 21

### PR DESCRIPTION
**Pull request by Malax**
Add support for Java 21. Changelog updated for immediate release.

Ref: GUS-W-14144442

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Main
 
+## v147
+
+* Add support for Java 21. ([#000](https://github.com/heroku/heroku-buildpack-jvm-common/pull/000))
```
Comment (edmorley) [inline CHANGELOG.md:7]:
```suggestion
* Add support for Java 21. ([#276](https://github.com/heroku/heroku-buildpack-jvm-common/pull/276))
```

File: lib/jvm.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@ DEFAULT_JDK_VERSION="1.8"
 DEFAULT_JDK_1_8_VERSION="1.8.0_382"
 DEFAULT_JDK_11_VERSION="11.0.20.1"
 DEFAULT_JDK_17_VERSION="17.0.8.1"
-DEFAULT_JDK_20_VERSION="20.0.2"
+DEFAULT_JDK_20_VERSION="21.0.0"
```
Comment (edmorley) [inline lib/jvm.sh:8]:
```suggestion
DEFAULT_JDK_21_VERSION="21.0.0"
```

File: lib/jvm.sh [hunk 3]
```diff
@@ -70,6 +71,7 @@ get_full_jdk_version() {
   "18") echo "${DEFAULT_JDK_18_VERSION}" ;;
   "19") echo "${DEFAULT_JDK_19_VERSION}" ;;
   "20") echo "${DEFAULT_JDK_20_VERSION}" ;;
+  "21") echo "${DEFAULT_JDK_20_VERSION}" ;;
```
Comment (edmorley) [inline lib/jvm.sh:74]:
```suggestion
  "21") echo "${DEFAULT_JDK_21_VERSION}" ;;
```

Comment (edmorley) [reply lib/jvm.sh:74]: Should the tests be tweaked to catch this? Currently they didn't fail even with the typo.

**Review by edmorley — APPROVED** [review]

File: lib/jvm.sh (hunk 3 above)
Comment (Malax) [reply lib/jvm.sh:74]: Yeah, that's super bad. The tests test that Java is installed and all the checks pass - but not if the correct version is installed. 🤔

Comment (Malax) [reply lib/jvm.sh]: Tracking here: https://github.com/heroku/heroku-buildpack-jvm-common/issues/277

**Pull request merged**